ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -375,10 +364,7 @@ private[platform] object MutableCacheBackedContractStore {
               maxBackoff = 10.seconds,
               randomFactor = 0.2,
             )
-          )(() =>
-            subscribeToContractStateEvents(contractStore.cacheIndex())
-              .map(contractStore.push)
-          )
+          )(() => subscribeToContractStateEvents().map(contractStore.push))
```

Comment:
just wondering: would it be the same if we do the mapping on the RestartSource instead? (next line)

Reply:
Not exactly the same: the restart wouldn't handle anymore errors that might arise from pushing to the contractStore. But that's fine, exceptions thrown there are unexpected, so I guess it's fine to tear down the system in that case.

Good suggestion. Changed